File: 프로그래머스/Lv.0/181929. 원소들의 곱과 합/원소들의 곱과 합.java
import java.util.Arrays;

class Solution {
    public int solution(int[] num_list) {
        
        int answer = -1;

        int sum = Arrays.stream(num_list).sum();

        int multiply = Arrays.stream(num_list).reduce(1, (a, b) -> a * b);

        if (multiply < sum * sum) {
            answer = 1;
        } else if (multiply > sum * sum) {
            answer = 0;
        }

        return answer;
    }
}

File: 프로그래머스/Lv.0/181929. 원소들의 곱과 합/원소들의 곱과 합.kt
class Solution {
    fun solution(num_list: IntArray): Int {
        var multiply = 1
        var sum = 0

        for (i in num_list) {
            multiply *= i
            sum += i
        }
        sum *= sum

        return if ((multiply < sum)) 1 else 0
    }
}